accepted/00000-snap-package-support.md

```diff
@@ -0,0 +1,483 @@
+- Feature Name: support snap packages in uyuni
+- Start Date: 6/01/2025
+
+# Summary
+[summary]: #summary
+
+Introduce Snap support in Uyuni, allowing users to:
+
+Install, update, and remove Snap packages.
+
+Manage Snap channels (stable, candidate, beta, edge).
+
+Handle Snap package management in airgapped environments.
+
+# Motivation
+[motivation]: #motivation
+
+Since Ubuntu 22.04 LTS, the Snap package technology has gained significant traction, and with the Ubuntu 24.04 LTS release, many deb packages are being migrated to Snap.
+
+With this increasing adoption of Snap packages in Ubuntu, Uyuni lacks native support for managing Snap-based applications. This creates a gap in package management, especially as more software is distributed via Snap.
+
+
+# Detailed design
+[design]: #detailed-design
+
+## Data Analsis 
+
+Total unique packages: 6564 (6/11/2025)
+Total stable packages: 6564
+Architecture count in stable packages:
+  - amd64: 6463
+  - armhf: 45
+  - arm64: 58
+  - all: 101
+  - i386: 67
+  - ppc64el: 10
+  - s390x: 10
+
+Total unique packages checked: 13746
+- Total in 'beta': 1677
+- Total in 'candidate': 1782
+- Total in 'edge': 3740
+
+Total size: 493 GB  around 500GB
+Average package size: 76.82 MB
+
+## Detailed Design
+
+Proposed approach:
+
+1. Load the full list of available packages.
+
+2. Allow the client to select which packages they want to manage.
+
+3. Download only the selected packages to the repository cache. 
```

Comment:
This is different from current channel management, where whole repositories are mirrored. Have you considered syncing all 6500 snaps instead of only the ones specified by a user?